docs/testplan/PMON-Chassis-Enhancements-test-plan.md

```diff
@@ -0,0 +1,672 @@
+                   Name        Description     	Slot    Oper-Status    Admin-Status
+                -------------  -----------------  ------  -------------  --------------
+                LINE-CARD1  imm36-400g-qsfpdd       1       Online           up
+      ```
+ *  ``` show chassis-modules status ``` report card status unchanged after shutdown on line card
+ 
+
+### 1.3 Check platform chassis midplane status 
+
+#### Steps
+ * Use command ` show chassis-modules midplane-status ` to get status of midplane in modular chassis
+
+#### Verify in
+ * Supervisor
+ * Line Card
+
+#### Sample Output
+     ```
+       on Supervisor:
+         show chassis-modules midplane-status
+              Name    IP-Address    Reachability
+        ----------  ------------  --------------
+        LINE-CARD0      10.0.0.1            True
+        LINE-CARD1      10.0.0.2           False
+        LINE-CARD2      10.0.0.3            True
+        LINE-CARD3      10.0.0.4            True
+    ```  
+     ```
+          on line card: 
+                     Name    IP-Address    Reachability
+                -----------  ------------  --------------
+                SUPERVISOR0     10.0.0.16            True
+
+     ```
+     
+
+#### Pass/Fail Criteria
+ * Verify output on supervisor lists all the line cards, midplane ip addresses and reachability status as expected  
+ * Verify on each Line Card supervisor reachability is correct and supervisor ip address is correct
+
+### 1.4 Check thermal sensor output in chassis state db 
+#### Steps
+ * Run command “redis-dump  -p 6380 -d 13 -y -k "*TEMP*" "on Supervisor
+ * Run command “redis-dump -d 6 -y -k "*TEMP*" on line card
+
+#### Verify in
+ * Supervisor
+ * Line Card
+ 
+#### Sample Output
+```
+ "TEMPERATURE_INFO_8|Thermal 7": {
+    "expireat": 1605120521.5682561, 
+    "ttl": -0.001, 
+    "type": "hash", 
+    "value": {
+      "critical_high_threshold": "N/A", 
+      "critical_low_threshold": "N/A", 
+      "high_threshold": "105.0", 
+      "low_threshold": "0.0", 
+      "maximum_temperature": "127.0", 
+      "minimum_temperature": "-2.0", 
+      "temperature": "47.0", 
+      "timestamp": "20190217 16:03:13", 
+      "warning_status": "False"
+    }
+
+```
+#### Pass/Fail Criteria
+ * Verify on the supervisor, thermal sensor data for itself and all applicable peripherals. Also verify:
+    * High threshold is greater than maximum_temperature.
+    * Low threshold is lesser than the minimum_temperature.
+    * Warning status is False except check criteria high or low threshold above is false
+ * Verify on each line card that all sensor data is correct. Also verify:
+    * High threshold is greater than maximum_temperature.
+    * Low threshold is lesser than the minimum_temperature.
+    * Warning status is False except check criteria high or low threshold above is false
+ 
+ 
+### 1.5 Check power budget output in chassis state db 
+
+#### Steps
+ * Run command “redis-dump -d 6 -y -k "*power*"” 
```

Comment:
Suggest surrounding commands with single backticks rather than quotes to produce inline code.